qa/rpc-tests/test_framework/mininode.py

```diff
@@ -121,6 +121,25 @@ def deser_compact_size(f):
         nit = struct.unpack("<Q", f.read(8))[0]
     return nit
 
+def ser_varint(n):
+    r = b""
+    while True:
+        r += struct.pack("B", (n & 0x7F) | (0x80 if len(r) > 0 else 0x00))
+        if n <= 0x7f:
+            break
+        n = (n >> 7) - 1;
+    return r;
+
+def deser_varint(f):
+    n = 0
+    while True:
+        c = struct.unpack("B", f.read(1))[0]
+        n = (n << 7) | (c & 0x7F)
+        if c & 0x80:
+            n += 1
+        else:
+            return n
```

Comment:
```Python
>>> x = ser_varint(128)
>>> y = deser_varint(BytesIO(x))
>>> y
0
```

Reply:
Actually,
```Python
>>> deser_varint(BytesIO(ser_varint(129)))
1
>>> deser_varint(BytesIO(ser_varint(130)))
2
>>> deser_varint(BytesIO(ser_varint(131)))
3
```

Reply:
Maybe I'm not using `BytesIO` correctly.. digging.

Reply:
Btw, no ; in python (L130-131).

Reply:
So, I think the big issue here is that the varints are actually calculated and then written *in the opposite order*.

Reply:
Okay, changing `ser_varint` last line to
```Python
return r[::-1]
```
fixes it.

Reply:
Fixed. Thanks for finding the root error.